internal/dag/builder.go

```diff
@@ -65,6 +70,11 @@ func (b *Builder) Build() *DAG {
 		StatusCache:        status.NewCache(gatewayNSName, gatewayController),
 	}
 
+	if b.Metrics != nil {
```

Comment:
This was the easiest way to not have to fix a bunch of tests; alternately, can create a wrapper struct that handles the metrics part like is done for the xds cache update metrics, but it requires switching a bunch of stuff to interfaces and this seemed simpler. Let me know if anyone prefers otherwise.